### DISCUSSION TITLE
Callout vs. Alert vs. Empty State + CTA

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#3836] [General] [answered]
Hi team!

Working through a dbt flow where we need to let a user know that before they can set up a dbt model, they need to have a warehouse connected. We worked through some options [here](https://www.figma.com/file/Sip1fVnZgdbAHCLnegRH73/%5BDesign%5D-dbt-%26-Git-Sync?type=design&node-id=420-46290&mode=design&t=IquyUpso4o9ZJWlM-4) and ultimately decided that the best UX is to allow a user to click on the "set up" button in the card, and then we can surface a modal:

![Screenshot 2024-03-27 at 1 14 47 PM](https://github.com/twilio-labs/paste/assets/87141551/d492c859-5633-478e-9c92-36f4ce4ceaad)

or the [latest thinking](https://www.figma.com/file/Sip1fVnZgdbAHCLnegRH73/%5BDesign%5D-dbt-%26-Git-Sync?type=design&node-id=316-97206&mode=design&t=IquyUpso4o9ZJWlM-4), which is to surface an alert/callout/empty state to route them to the connecting a warehouse flow. Big question is: can we add a button in the alert?

Happy to chat through this further in office hours, thanks!
Colleen

**Comment by serifluous** [comment, marked as answer]
Hey @ceyerkes ! Yep, you can add a button in an Alert or Callout, just like in the Engage Audiences example that Charmaine (I think) pasted in your Figma file. We've even suggested using a small Button in the past, so it's a little more proportional to being in an Alert, but yeah, you can put any content there.

Out of curiosity, why is the latest thinking using the alert + empty state option, instead of including it as part of the extension set-up flow?

**Reply by ceyerkes** [reply]
thank you!

we considered including warehouse setup as part of the dbt connection process if one wasn't setup yet, but ultimately decided to steer users over to the actual flow instead of integrating into the dbt wizard to keep things a little cleaner. open to feedback on this though!
